src/screens/DiscoveryScreenV2/elements/Search.js

```diff
@@ -214,9 +215,27 @@ const DiscoverySearch = ({
   );
 };
 
+DiscoverySearch.propTypes = {
+  setDiscoveryLoadingData: PropTypes.func,
+  searchText: PropTypes.string,
+  setSearchText: PropTypes.func,
+  placeholderText: PropTypes.string,
+  setIsFocus: PropTypes.func,
+  setIsFirstTimeOpen: PropTypes.func,
+  fetchDiscoveryData: PropTypes.func,
+  fetchData: PropTypes.func,
+  onCancelToken: PropTypes.func,
+  hideBackIcon: PropTypes.bool,
+  autoFocus: PropTypes.bool
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Defining propTypes for the component is a good practice as it helps to validate the props being received by the component. It also serves as documentation for the component's API.
```diff
+ DiscoverySearch.propTypes = {
+   setDiscoveryLoadingData: PropTypes.func,
+   searchText: PropTypes.string,
+   setSearchText: PropTypes.func,
+   placeholderText: PropTypes.string,
+   setIsFocus: PropTypes.func,
+   setIsFirstTimeOpen: PropTypes.func,
+   fetchDiscoveryData: PropTypes.func,
+   fetchData: PropTypes.func,
+   onCancelToken: PropTypes.func,
+   hideBackIcon: PropTypes.bool,
+   autoFocus: PropTypes.bool
+ };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->